src/internal/connector/exchange/cache.go

```diff
@@ -0,0 +1,136 @@
+package exchange
+
+import (
```

Comment:
mmm I'd slightly prefer `container_resolver` I suppose? In hindsight just `cache` is rather vague and `lookup` (or `container_lookup`) doesn't quite match the functionality since it is storing values locally for later use